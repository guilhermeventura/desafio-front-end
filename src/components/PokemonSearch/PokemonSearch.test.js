import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import renderer from "react-test-renderer";

import { default as PokemonSearch } from "./index";

describe("<PokemonSearch />", () => {
  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  const store = mockStore({ pokemons: [] });
  test("renders correctly", () => {
    const component = renderer
      .create(
        <Provider store={store}>
          <PokemonSearch />
        </Provider>
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
