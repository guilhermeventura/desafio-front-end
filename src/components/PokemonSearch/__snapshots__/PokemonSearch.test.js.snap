// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonSearch /> renders correctly 1`] = `
<div
  className="columns is-multiline is-centered"
  style={Object {}}
>
  <div
    className="column is-half"
    style={Object {}}
  >
    <h3
      className="mdc-typography mdc-typography--headline3"
    >
      Search a pokemon below
    </h3>
    <br />
    <div
      className="mdc-text-field mdc-text-field--outlined mdc-text-field--with-trailing-icon"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <input
        className="mdc-text-field__input"
        disabled={false}
        id=""
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        value=""
      />
      <div
        className="mdc-notched-outline mdc-notched-outline--no-label"
      >
        <div
          className="mdc-notched-outline__leading"
        />
        <div
          className="mdc-notched-outline__notch"
          style={
            Object {
              "width": undefined,
            }
          }
        >
          <label
            className="mdc-floating-label"
            htmlFor=""
            onAnimationEnd={[Function]}
          >
            Name or number...
          </label>
        </div>
        <div
          className="mdc-notched-outline__trailing"
        />
      </div>
      <i
        className="material-icons mdc-text-field__icon"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        search
      </i>
    </div>
  </div>
</div>
`;
